### PR TITLE
Update eslint to 8.36.0

### PR DESCRIPTION
**Pull request by yashincontrol**
After the PRs on appui (https://github.com/iTwin/appui/pull/216) and presentation (https://github.com/iTwin/presentation/pull/89) are merged, there should be no issues with this update. Here is nightman for reference: https://github.com/iTwin/nightman/actions/runs/4500732451/jobs/7920240470#step:16:110

### CODE REVIEW

**Review by ben-polinsky — APPROVED** [review]

**Review by aruniverse — CHANGES_REQUESTED** [review]
beachball change?

**Review by aruniverse — APPROVED** [review]

File: change/@itwin-eslint-plugin-9e5d024e-f1c6-4768-b61b-f5eacd8a2d26.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply change/@itwin-eslint-plugin-9e5d024e-f1c6-4768-b61b-f5eacd8a2d26.json:2]: when we're ready to do an actual release itll be major

[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply change/@itwin-eslint-plugin-9e5d024e-f1c6-4768-b61b-f5eacd8a2d26.json:2]: but the whole point of beachball is to track which changes require a major bump. You shouldn't manually override the bump type, this file is what will define which version gets bumped

[the remark this answers is not in the document]
Comment (aruniverse) [reply change/@itwin-eslint-plugin-9e5d024e-f1c6-4768-b61b-f5eacd8a2d26.json:2]: fair.

@yashincontrol change this to major

[the remark this answers is not in the document]
Comment (yashincontrol) [reply change/@itwin-eslint-plugin-9e5d024e-f1c6-4768-b61b-f5eacd8a2d26.json:2]: Okay

**Review by paulius-valiunas — APPROVED** [review]

**Comment by yashincontrol** [comment]
Update: The above PRs have been merged and nightman looks good https://github.com/iTwin/nightman/actions/runs/4510910957/jobs/7942425243

**Pull request merged**